frontend/scripts/gameboard.js

```diff
@@ -26,53 +26,67 @@ for (let i = 0; i<21; i++){
 }
 
 function drawTile(row,col){
-    console.log('row', row)
-    console.log('col', col)
     ctx.fillRect(Number(col)*60+1, Number(row)*60+1, 58, 58)
+
 }
 
 function drawRandomPiece() {
     const pieces = [iPiece, jPiece, lPiece, oPiece, zPiece, tPiece, sPiece]
     const RandomNum = Math.floor(Math.random() * pieces.length)
+    // return testPiece;
     return pieces[RandomNum]
     //console.log(pieces[RandomNum])
      
 }
 
 let column = 0;
 let piece = drawRandomPiece();
+
 function drawPiece(piece) {
     ctx.fillStyle = piece.color
     const array = piece.array;
-    console.log(piece.color)
     for (const rowIndex in array){
         const row = array[rowIndex]
         for (const colIndex in row) {
             if (row[colIndex] === 1) {
-                drawTile(Number(rowIndex),Number(colIndex) + column)
+                drawTile(Number(rowIndex),Number(colIndex) + column) 
             
             }
         }
     }
 
 }
 
+
 function emptyGrid(){
-    console.log()
-    console.log()
     ctx.fillStyle = "white";
     for(let x = 0; x < 10; x++ ){
         for(let y = 0; y < 20; y++){
-            ctx.fillRect(x*60+1, y*60+1, 58, 58)
-                
-            
-            
+            ctx.fillRect(x*60+1, y*60+1, 58, 58)   
         }    
     }
 }
 
+function isMoveValid(newColumn) {
+    for (let rowi = 0; rowi < piece.array.length; rowi++) {
+        const row = piece.array[rowi];
+        for (let coli = 0; coli < row.length; coli++) {
+            const element = row[coli];
+            console.log(newColumn + coli)
+            if (element === 1 && newColumn + coli == 10 || newColumn + coli == -1) { // cheeky little or value instead of another if statement ;)
```

Comment:
I like the thought of this, but I dont think this actually works in all cases. Essentially the && operator has higher priority that the || operator. Sow hat you have written is equivalent to:

` (element === 1 && newColumn + coli == 10) || newColumn + coli == -1`

ie when we are checking the left boundary, we are no longer checking that element === 1.

We can fix this by either splitting this out into a new if statement, or by replacing the conditional with the following:
`if ((element === 1 && newColumn + coli == 10) || (element === 1 && newColumn + coli == -1))`